src/codegen/doc/format.rs

```diff
@@ -0,0 +1,59 @@
+use regex::{Captures, Regex};
+
+const LANGUAGE_SEP_END : &'static str = "\" -->";
+const LANGUAGE_BLOCK_BEGIN : &'static str = "|[<!-- language=\"";
+const LANGUAGE_BLOCK_END : &'static str = "\n]|";
+
+pub fn reformat_doc(input: &str) -> String {
+    let out = code_blocks_transformation(input);
+    replace_c_types(&out)
+}
+
+fn code_blocks_transformation(input: &str) -> String {
+    let mut out = String::new();
+
+    for entry in input.split(LANGUAGE_BLOCK_BEGIN) {
+        if out.is_empty() {
+            out.push_str(entry);
```

Comment:
Although I'm not quite sure what's going to happen if there are two code blocks.